### PR TITLE
[BEAM-1827] Fix use of deprecated Spark APIs in the runner.

### PR DESCRIPTION
**Pull request by amitsela**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by amitsela** [comment]
R: @aviemzur

**Comment by amitsela** [comment]
Run Spark RunnableOnService

**Comment by amitsela** [comment]
Run Spark ValidatesRunner

File: runners/spark/src/main/java/org/apache/beam/runners/spark/translation/streaming/SparkRunnerStreamingContextFactory.java [hunk 1]
```diff
@@ -52,54 +53,59 @@
...
-      SparkPipelineOptions options,
-      CheckpointDir checkpointDir) {
+      Pipeline pipeline, SparkPipelineOptions options, CheckpointDir checkpointDir) {
     this.pipeline = pipeline;
     this.options = options;
     this.checkpointDir = checkpointDir;
   }
 
-  private EvaluationContext ctxt;
+  public Function0<JavaStreamingContext> create() {
```
Comment (aviemzur) [inline runners/spark/src/main/java/org/apache/beam/runners/spark/translation/streaming/SparkRunnerStreamingContextFactory.java]: Can this be replaced by a `public static` member? It is already lazy thanks to `Function0#call`.

**Comment by amitsela** [comment]
Run Spark ValidatesRunner

**Comment by amitsela** [comment]
@aviemzur PTAL I've made `SparkRunnerStreamingContextFactory` simply implement `Function0` but I had to make its members transient or findbugs would get mad 😠

**Comment by amitsela** [comment]
Run Spark ValidatesRunner

**Comment by aviemzur** [comment]
LGTM pending tests.

**Pull request closed without merging**